### ISSUE TITLE
Switch htmLawed to ezyang/htmlpurifier

### ISSUE THREAD

**Issue by edgardmessias** [glpi-project/glpi#1920]
Switch htmLawed to ezyang/htmlpurifier

Differences: http://htmlpurifier.org/comparison#Summary

**Comment by trasher** [comment]
According to the page you've linked; htmlpurifier from 2017 is compared against very old releases of all other libraries.

There seems to be a 2017 release of htmlLawed; that comparison should be updated with latest informations.

**Comment by trasher** [comment]
In GLPI; the version od htmlawer we use is 1.1.21 from 2016-02-27

**Comment by tomolimo** [comment]
@Trasher, could you just give me some reasons why to use such libraries when PHP provides also some mechanisms to clean HTML (like http://php.net/manual/fr/function.strip-tags.php)?

**Comment by trasher** [comment]
@tomolimo no. I'm not the one which introduces the existing lib. As far as I can see, it does more than just `strip_tags`; but I'm not going to spend time on my side right now to get a full speech about that. I'm just answering on the lib change.

**Comment by tomolimo** [comment]
@Trasher ok, thank you

**Comment by trasher** [comment]
I do not see any benefit to the project to switch that lib right now, I'm closing.

If you have any arguments to produce, feel free to reopen :)

**Comment by trasher** [comment]
Hum... It appears that htmlaweed is not present on packagist and thus cannot be imported as all other dependencies in the composer file.

This is a pro for a change :) Reopening.

**Comment by trasher** [comment]
As @remicollet pointed out; we need to improve current system unit tests before switching.

**Comment by edgardmessias** [comment]
@trasher ,about unit tests, you can try http://codeception.com/
